jax/_src/lax/parallel.py

```diff
@@ -534,14 +536,14 @@ def _ppermute_transpose_rule(t, perm, axis_name):
   inverse_perm = list(zip(dsts, srcs))
   return [ppermute(t, axis_name=axis_name, perm=inverse_perm)]
 
-def _ppermute_batcher(vals_in, dims_in, axis_size, axis_name, perm):
-  assert len(perm) == axis_size, "Permutation doesn't match the axis size!"
-  perm_indices = np.full((axis_size,), -1, dtype=np.int32)
-  for s, d in perm:
-    perm_indices[s] = d
-  vals_out = [lax_numpy.take(v, perm_indices, d) if d is not batching.not_mapped else v
-              for v, d in zip(vals_in, dims_in)]
-  return vals_out, dims_in
+def _ppermute_batcher(frame, vals_in, dims_in, axis_name, perm):
```

Comment:
This should check that `axis_name == frame.name` or otherwise it will silently compute something incorrect.

Reply:
I think this should only be called when `axis_name == frame.name`, but I'll add an assertion. LMK if that sounds incorrect!

Reply:
I guess it's analogous to how `_batched_reduction_collective` includes `assert frame.name in axis_name`.